lab_5_scrapper/scrapper.py

```diff
@@ -1,164 +1,333 @@
+        self.config = config
+        self.article = Article(url=self.full_url, article_id=self.article_id)
 
     def _fill_article_with_text(self, article_soup: BeautifulSoup) -> None:
         """
         Finds text of article
         """
-        pass
+        content_body = article_soup.find('section', {'class': 'article-body'})
+        paragraphs_body = content_body.find_all('p')
+        text_body = ''.join(i.text.strip() for i in paragraphs_body)
+        self.article.text = text_body
 
     def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> None:
         """
         Finds meta information of article
         """
-        pass
+        title = article_soup.find('h1', {'class':'article-head__title'})
+        if title:
+            self.article.title = title.text
+        info = article_soup.find('div', {'class':'article-head__info'})
+        authors = info.find_all('a')
+        authors_list = []
+        for author in authors:
+            author_cleared = [str(tag) for tag in author][0].strip()
+            authors_list.append(author_cleared)
+        self.article.author = authors_list
+        self.article.date = self.unify_date_format(date_str=str(info.span.get('date-time')))
+        tags_list = []
+        tags = article_soup.find_all('div', {'class':'article-footer__hashtags-item'})
+        if tags:
+            for tag in tags:
+                tags_list.append(tag.text.strip())
+            self.article.topics = tags_list
+        else:
+            self.article.topics = ['none']
 
     def unify_date_format(self, date_str: str) -> datetime.datetime:
         """
         Unifies date format
         """
-        pass
+        return datetime.datetime.strptime(date_str, '%d.%m.%Y %H:%M:%S')
 
     def parse(self) -> Union[Article, bool, list]:
         """
         Parses each article
         """
-        pass
+        response = make_request(url=self.full_url, config=self.config)
+        soup = BeautifulSoup(response.text, 'html.parser')
+        self._fill_article_with_text(article_soup=soup)
+        self._fill_article_with_meta_information(article_soup=soup)
+        return self.article
 
 
 def prepare_environment(base_path: Union[Path, str]) -> None:
     """
     Creates ASSETS_PATH folder if no created and removes existing folder
     """
-    pass
+    path_for_environment = Path(base_path)
+
+    if path_for_environment.exists():
+        shutil.rmtree(path_for_environment)
+    path_for_environment.mkdir(parents=True)
 
 
 def main() -> None:
     """
     Entrypoint for scrapper module
     """
-    pass
+    config = Config(CRAWLER_CONFIG_PATH)
+    prepare_environment(ASSETS_PATH)
+    crawler = Crawler(config=config)
+    crawler.find_articles()
+    for i, url in enumerate(crawler.urls, start=1):
+        parser = HTMLParser(full_url=url, article_id=i, config=config)
+        article = parser.parse()
+        if isinstance(article, Article):
```

Comment:
why do you specify this condition?

Reply:
> why do you specify this condition?

This is due to the fact that the to_meta() and to_raw() functions take strictly the Article class as input, and the parse() function outputs Union[Article, bool, list]. With this condition, we formally restrict the possible output to the Article class.